.github/workflows/publish-site.yml

```diff
@@ -32,9 +37,12 @@ jobs:
         run: yarn --immutable
       - name: Run build script
         run: yarn build
-      - name: Deploy to `${{ inputs.destination_dir }}` directory of `gh-pages` branch
-        uses: peaceiris/actions-gh-pages@de7ea6f8efb354206b205ef54722213d99067935
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v2
         with:
-          personal_token: ${{ secrets.METAMASKBOT_TOKEN }}
-          publish_dir: ./public
-          destination_dir: ${{ inputs.destination_dir }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: ghactionssession
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Deploy to `${{ inputs.destination_dir }}` directory of remote storage
+        run: |
+          aws s3 cp ./public s3://${{ secrets.AWS_BUCKET_NAME }}/${{ inputs.destination_dir }} --recursive --acl private
```

Comment:
Will this still provide access to previous versions?

Old version are still sometimes used, and we need the version in the URL to set-up client-side caching properly, see that current deployments have version inside the URL - https://metamask.github.io/iframe-execution-environment/0.14.0/

Reply:
This is only meant for release a new version, If we want to provide access to previous versions from the new URL we would need to add them which shouldn't be a problem.

Reply:
The minimum requirement is that the version has to be in the URL. Optimal requirement is that previous version are also accessible.